### PR TITLE
Persist the Fyne global settings

### PR DESCRIPTION
**Pull request by andydotxyz**
Watch the file (directory) for changes and re-load if it changes.
This is the basis of allowing a global settings storage that can be saved with a future settings UI to manage it.

### CODE REVIEW

File: app/settings.go [hunk 1]
```diff
@@ -50,8 +71,74 @@ func (s *settings) apply() {
...
+		return err
+	}
+	decode := json.NewDecoder(file)
+
+	return decode.Decode(&s.schema)
+}
+
+func (s *settings) setupTheme() {
+	name := s.schema.ThemeName
+	env := os.Getenv("FYNE_THEME")
```
Comment (toaster) [inline app/settings.go]: How about:

```go
if envTheme := os.Getenv("FYNE_THEME"); envTheme != nil {
  …
}
```

**Comment by toaster** [comment]
I don't want to be dogmatic about this but there are no automatic tests :(.

File: app/settings.go (hunk 1 above)
Comment (andydotxyz) [reply app/settings.go]: then envTheme is not in scope for the code below surely?

**Comment by andydotxyz** [comment]
Quite right. For some reason the test files were not committed - let me look for them

**Comment by andydotxyz** [comment]
It wasn't much but I found the missing file :)

**Comment by andydotxyz** [comment]
OK this branch is a bit of a work in progress whilst I refactor some things and get it ready to have the settings UI published as part of the work... I'll post here shortly when it's ready for re-review

File: app/settings.go (hunk 1 above)
Comment (andydotxyz) [reply app/settings.go]: Oh sorry, I see it now :)

**Comment by andydotxyz** [comment]
This is ready to re-review now. It has more tests, a better settings UI and the ability to globally set scale (or use auto)

**Comment by andydotxyz** [comment]
Oops, didn't realise the merge would invalidate the review.
The merge was required for mod metadata conflict but I think we're good.

**Pull request merged**